x-pack/plugins/ml/public/plugin.ts

```diff
@@ -164,25 +164,24 @@ export class MlPlugin implements Plugin<MlPluginSetup, MlPluginStart> {
       }).enable();
     }
 
+    if (pluginsSetup.home) {
+      registerFeature(pluginsSetup.home);
+    }
+
     const licensing = pluginsSetup.licensing.license$.pipe(take(1));
     licensing.subscribe(async (license) => {
       const mlEnabled = isMlEnabled(license);
-      const fullLicense = isFullLicense(license);
-      const [coreStart, pluginStart] = await core.getStartServices();
-      const { capabilities } = coreStart.application;
-
-      if (mlEnabled) {
-        // add ML to home page
-        if (pluginsSetup.home) {
-          registerFeature(pluginsSetup.home);
```

Comment:
There was no guarantee that the home app hadn't already called `featureCatalogue.get()` before this statement could run, making ML not present in the effective feature catalogue.